src/backend/src/controllers/change-requests.controllers.ts

```diff
@@ -76,8 +71,101 @@ export const reviewChangeRequest = async (req: Request, res: Response) => {
         approved: true
       }
     });
-  }
+    const wbs = await prisma.wBS_Element.findUnique({
+      where: {
+        wbsElementId: foundCR.wbsElementId
+      },
+      include: {
+        workPackage: true,
+        project: true
+      }
+    });
+
+    if (!wbs) {
+      return res.status(404).json({ message: `wbs element with id #${foundCR.wbsElementId} not found` });
+    }
+
+    const wp = { ...wbs.workPackage };
+
+    const proj = { ...wbs.project };
+
+    if (!wp && proj) {
+      let updatedBudgetImpact;
+      if (!proj.budget) {
+        updatedBudgetImpact = 0 + foundPs.budgetImpact;
+      } else {
+        updatedBudgetImpact = proj.budget + foundPs.budgetImpact;
+      }
+      const change = {
+        changeRequestId: crId,
+        implementerId: reviewerId,
+        detail: buildChangeDetail('Budget Impact', String(proj.budget), String(updatedBudgetImpact))
+      };
+      await prisma.project.update({
+        where: { projectId: wbs.projectNumber },
+        data: {
+          budget: updatedBudgetImpact,
+          wbsElement: {
+            update: {
+              changes: {
+                createMany: {
+                  data: [change]
+                }
+              }
+            }
+          }
+        }
+      });
+    } else if (wp) {
+      const wpProj = await prisma.project.findUnique({
+        where: { projectId: wp.projectId }
+      });
+      if (!wpProj) {
+        return res.status(400).json({ message: 'Work package project not found' });
+      }
+      const updatedBudgetImpact = wpProj.budget + foundPs.budgetImpact;
+      let updatedDuration;
+      if (!wp.duration) {
```

Comment:
wp should always have duration

Reply:
This is similar to before where it says its an optional value for some reason so I changed it to a force unwrap